src/test/java/mil/dds/anet/test/resources/AnetEmailResourceTest.java

```diff
@@ -0,0 +1,65 @@
+package mil.dds.anet.test.resources;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.dropwizard.testing.junit5.DropwizardAppExtension;
+import mil.dds.anet.AnetObjectEngine;
+import mil.dds.anet.config.AnetConfiguration;
+import mil.dds.anet.emails.ReportEmail;
+import mil.dds.anet.test.TestData;
+import mil.dds.anet.test.client.AnetBeanList_Report;
+import mil.dds.anet.test.client.ReportSearchQueryInput;
+import mil.dds.anet.threads.AnetEmailWorker;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class AnetEmailResourceTest extends AbstractResourceTest {
+  @Autowired
+  protected DropwizardAppExtension<AnetConfiguration> dropwizardApp;
+  private AnetEmailWorker emailWorker;
+
+  @BeforeAll
+  void setUpClass() {
+    final AnetObjectEngine engine = AnetObjectEngine.getInstance();
+    emailWorker = new AnetEmailWorker(dropwizardApp.getConfiguration(), engine.getEmailDao());
+    emailWorker.run();
+  }
+
+  @AfterAll
+  void tearDownClass() {
+    emailWorker.run();
+    emailWorker = null;
+    AnetEmailWorker.setInstance(null);
+  }
+
+  @Test
+  void shouldReturnEmailData() {
+    final var reportQuery = ReportSearchQueryInput.builder()
+        .withAuthorPositionUuid(admin.getPosition().getUuid()).build();
+
+    // Search by author position
+    final AnetBeanList_Report results = withCredentials(adminUser,
+        t -> queryExecutor.reportList(getListFields("{ uuid }"), reportQuery));
+    final var reportUuid = results.getList().iterator().next().getUuid();
+    final var input = TestData.createAnetEmailInput();
+    final var mutationResult =
+        withCredentials(adminUser, t -> mutationExecutor.emailReport("", input, reportUuid));
+    assertThat(mutationResult).isEqualTo(1);
+
+    final var pendingEmails = withCredentials(adminUser,
+        t -> queryExecutor.pendingEmails("{ id toAddresses createdAt comment errorMessage type }"));
+    assertThat(pendingEmails).hasSize(1)
+        .allSatisfy(pendingEmail -> assertThat(pendingEmail).usingRecursiveComparison()
+            .ignoringFields("__typename", "id", "createdAt", "type", "errorMessage")
```

Comment:
Why retrieve `"id", "createdAt", "type", "errorMessage"` if you're ignoring them anyway? You could just retrieve the ones you're interested in (`toAddresses comment`) and remove the call to `ignoringFields()`.

Reply:
It validates the query itself. If the fields were not available, this would result in a GraphQL Error.

Reply:
To me it feels a bit pointless to check the GraphQL API in that way. In any case the `__typename` field can be removed from your list. And you could at least assert something on the `type`.

Reply:
Point taken on the `_typename`. There's an assertion for `type` two lines further down.

Reply:
Ah, I see, you're comparing an `AnetEmailInput` object to an `AnetEmail` object, and they have a different `__typename`…

Reply:
You could use:
```java
    assertThat(pendingEmails).hasSize(1)
        .allSatisfy(pendingEmail -> assertThat(pendingEmail).usingRecursiveComparison()
            .comparingOnlyFields("toAddresses", "comment").isEqualTo(input))
        […]
```
which is IMHO clearer (and simpler).